ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc

```diff
@@ -711,21 +713,73 @@ TFLiteInterpreter::cacheInOutTensorPtr ()
 /**
  * @brief	TFLiteCore constructor
  */
-TFLiteCore::TFLiteCore ()
+TFLiteCore::TFLiteCore (const GstTensorFilterProperties *prop)
 {
   num_threads = -1;
   accelerator = ACCL_NONE;
   delegate = TFLITE_DELEGATE_NONE;
   interpreter_sub = nullptr;
-  interpreter = new TFLiteInterpreter ();
+  shared_tensor_filter_key = NULL;
+
+  if (prop->shared_tensor_filter_key) {
+    shared_tensor_filter_key =
+        g_strdup (prop->shared_tensor_filter_key);
+    checkSharedInterpreter (prop);
+  }
+  else
+    interpreter = new TFLiteInterpreter ();
 }
 
 /**
  * @brief	TFLiteCore destructor
  */
 TFLiteCore::~TFLiteCore ()
 {
-  delete interpreter;
+  if (shared_tensor_filter_key) {
+    if (nnstreamer_filter_shared_table_remove (shared_tensor_filter_key, this)) {
+      if (nnstreamer_filter_referred_list_is_empty (shared_tensor_filter_key)) {
```

Comment:
Possible thread errors.
Thread 1, tensor_filter_tensorflow_lite.cc LINE 739 (TRUE)
Thread 1, tensor_filter_tensorflow_lite.cc LINE 740 (TRUE)
Thread 2, tensor_filter_tensorflow_lite.cc LINE 767 (FALSE)
Thread 1. tensor_filter_tensorflow_lite.cc LINE 741
Thread 2.  tensor_filter_tensorflow_lite.cc ==> SEGMENTATION FAULT.

Reply:
nnstreamer_filter_referred_list_is_empty SHOULD BE NOT NEEDED.
nnstreamer_filter_shared_table_remove should be able to do "delete interpreter" atomically if needed. (defining a callback is fine.)
Otherwise, you'd need a mutex protecting an internal object from external logic (nnstreamer_filter_*), which doesn't look good (increasing coupling between the framework and its drivers).